staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go

```diff
@@ -615,6 +615,222 @@ func (s *store) GetList(ctx context.Context, key string, opts storage.ListOption
 		attribute.Int("limit", int(opts.Predicate.Limit)),
 		attribute.String("continue", opts.Predicate.Continue))
 	defer span.End(500 * time.Millisecond)
+	if opts.Recursive {
```

Comment:
I'm not sure what happened here.

The whole idea for having a common path for recursive and non-recursive was that these effectively share the same code. We had at least one bug that I remember (and probably more) that we fixed sth only in one of those path.

So I'm actually strongly against this change that is unnecessarily duplicating a bunch of code.

Reply:
Duplicating the code is a required step for splitting them. The code is so couple that untangling it will require making code more complicated before we can clean it. The question is whether you prefer to review everything at once in one PR or have smalls like this that might leave some things to be desired.

https://github.com/kubernetes/kubernetes/compare/master...serathius:kubernetes:split-tmp